src/pubsub/ua_pubsub_reader.c

```diff
@@ -277,20 +277,20 @@ UA_DataSetReader_generateNetworkMessage(UA_PubSubConnection *pubSubConnection, U
 
     switch(dataSetReader->config.publisherId.type->typeKind) {
     case UA_DATATYPEKIND_BYTE:
-        nm->publisherIdType = UA_PUBLISHERDATATYPE_BYTE;
-        nm->publisherId.publisherIdByte = *(UA_Byte *) dataSetReader->config.publisherId.data;
+        nm->publisherIdType = UA_PUBLISHERIDTYPE_BYTE;
+        nm->publisherId.byte = *(UA_Byte *) dataSetReader->config.publisherId.data;
         break;
     case UA_DATATYPEKIND_UINT16:
-        nm->publisherIdType = UA_PUBLISHERDATATYPE_UINT16;
-        nm->publisherId.publisherIdUInt16 = *(UA_UInt16 *) dataSetReader->config.publisherId.data;
+        nm->publisherIdType = UA_PUBLISHERIDTYPE_UINT16;
+        nm->publisherId.uint16 = *(UA_UInt16 *) dataSetReader->config.publisherId.data;
         break;
     case UA_DATATYPEKIND_UINT32:
-        nm->publisherIdType = UA_PUBLISHERDATATYPE_UINT32;
-        nm->publisherId.publisherIdUInt32 = *(UA_UInt32 *) dataSetReader->config.publisherId.data;
+        nm->publisherIdType = UA_PUBLISHERIDTYPE_UINT32;
+        nm->publisherId.uint32 = *(UA_UInt32 *) dataSetReader->config.publisherId.data;
         break;
     case UA_DATATYPEKIND_UINT64:
-        nm->publisherIdType = UA_PUBLISHERDATATYPE_UINT64;
-        nm->publisherId.publisherIdUInt64 = *(UA_UInt64 *) dataSetReader->config.publisherId.data;
+        nm->publisherIdType = UA_PUBLISHERIDTYPE_UINT64;
+        nm->publisherId.uint64 = *(UA_UInt64 *) dataSetReader->config.publisherId.data;
         break;
     default:
```

Comment:
String is missing.

Reply:
Thanks for review

With fast-path configuration string PublisherId was not supported for the Reader before, that's why I haven't changed it.
But it was supported for the Publisher ... and I haven't understood the reason, because even if string is dynamic the PublisherId is not a dynamic field which changes at runtime ...

UA_Server_freezeReaderGroupConfiguration() -> UA_DataSetReader_generateNetworkMessage()
I have removed the check
```
    if(!UA_DataType_isNumeric(dataSetReader->config.publisherId.type))
        return UA_STATUSCODE_BADNOTSUPPORTED;
```

checkReaderIdentifier() checks string PublisherIds

But subscribing fails with
`PubSub receive. Unknown field type.`

Reply:
I have a problem with fast-path Subscriber string PublisherId implementation:

- I couldn't remove the PublisherId String clear within `UA_NetworkMessage_clear()`, because the `UA_NetworkMessageHeader_decodeBinary` - `UA_String_decodeBinary` allocates memory at every receive() cycle

- therefore I also had to alloc memory (deep copy) at `UA_Server_freezeReaderGroupConfiguration()` - `UA_DataSetReader_generateNetworkMessage` because it uses `UA_NetworkMessage_clear()` at `unfreezeReaderGroupConfiguration`

- But `UA_NetworkMessage_updateBufferedNwMessage` allocates memory with `UA_String_decodeBinary()` into the same prepared (generated at freeze) publisherId string variable ... and in the worst case it uses buffer malloc and at teardown we have to call clear() because otherwise we would not clear the last received string

Reply:
If I would immediately clear the PublisherId String after UA_DataSetReader_generateNetworkMessage() after the size has been calculated then I would not have a problem during subscribe cycle, but at unfreezeReaderGroup I may not free anything
but there the same UA_NetworkMessage_clear() function is used as for subscribe Cycle ...
and then we would free invalid memory